File: examples/football.py
import sys

def load_results(filename):
    with open(filename, 'r') as f:
        raw = f.read()

    lines = raw.split('\n') # readlines support in v0.3
    return [parse_result(line) for line in lines if line]

def parse_result(line):
    away_index = line.index(' - ') + 3
    result_index = line.index(' ', away_index) + 1
    goals = line[result_index:].split(':')
    return line[:away_index - 3], line[away_index:result_index - 1], (int(goals[0]), int(goals[1]))

def calculate_points(results, team): # call(*arg) supported only for tuples
    return sum(result_points(team, *result) for result in results)

def result_points(team, host, away, goals):
    if host == team and goals[0] > goals[1] or away == team and goals[0] < goals[1]:
        return 3
    elif goals[0] == goals[1] and (host == team or away == team):
        return 1
    else:
        return 0

if len(sys.argv) < 3:
    print('usage: football <stats-file> <team>')
else:
    results = load_results(sys.argv[1])
    # print(result_points(sys.argv[2], *results[0]))
    print(calculate_points(results, sys.argv[2]))

File: examples/football.go
package main

import (
	"strings"
	"strconv"
	"fmt"
	"io/ioutil"
	"os"
)

func LoadResults(filename string) []Result {
	_contents, _ := ioutil.ReadFile(filename)
	raw := string(_contents)
	lines := strings.Split(raw, "\n")
	var _results []Result
	for _, line := range lines {
		if len(line) > 0 {
			_results = append(_results, ParseResult(line))
		}
	}
	return _results

}

func ParseResult(line string) *Result {
	awayIndex := strings.Index(line, " - ") + 3
	resultIndex := awayIndex + strings.Index(line[awayIndex:], " ") + 1
	goals := strings.Split(line[resultIndex:], ":")
	_int, _ := strconv.Atoi(goals[0])
	_int1, _ := strconv.Atoi(goals[1])
	return Result{line[:awayIndex - 3], line[awayIndex:resultIndex - 1], [...]int{_int, _int1}}
}

func CalculatePoints(results []Result, team string) int {
	accumulator := 0
	for _, result := range results {
		accumulator += ResultPoints(team, result)
	}

	return accumulator
}

func ResultPoints(team string, result Result) int {
	if result.Host == team && result.Goals[0] > result.Goals[1] || result.Away == team && result.Goals[0] < result.Goals[1] {
		return 3
	} else if result.Goals[0] == result.Goals[1] && (result.Host == team || result.Away == team) {
		return 1
	} else {
		return 0
	}
}

type Result struct {
	Host string
	Away string
	Goals [2]int
}

func main() {
	if len(os.Args) < 3 {
		fmt.Println("usage: football <stats-file> <team>")
	} else {
		results := LoadResults(os.Args[1])
		fmt.Println(CalculatePoints(results, os.Args[2]))
	}
}
